pkg/bufferdecoder/net_decoder.go

```diff
@@ -118,3 +129,74 @@ func (decoder *EbpfDecoder) DecodeNetDebugEvent(netDebugEvent *NetDebugEvent) er
 	decoder.cursor += int(netDebugEvent.GetSizeBytes())
 	return nil
 }
+
+type DnsQueryData struct {
+	Query      string `json:"query"`
+	QueryType  string `json:"queryType"`
+	QueryClass string `json:"queryClass"`
+}
+
+// parseDnsQuestion parse dns request to DnsQueryData
+func parseDnsQuestion(question layers.DNSQuestion) DnsQueryData {
+	var request DnsQueryData
+	request.Query = string(question.Name)
+	request.QueryType = question.Type.String()
+	request.QueryClass = fmt.Sprint("", question.Class)
+	return request
+}
+
+func (decoder *EbpfDecoder) DecodeDnsQueryArray(questions *[]DnsQueryData) error {
+	offset := decoder.cursor
+	packet := gopacket.NewPacket(decoder.buffer[offset:], layers.LayerTypeEthernet, gopacket.Default)
+	if packet == nil {
+		return fmt.Errorf("couldnt parse the packet")
+	}
+	dnsLayer := packet.Layer(layers.LayerTypeDNS).(*layers.DNS)
```

Comment:
I think that a cleaner way to do this is to use the following format:
```suggestion
	dnsLayer, ok := packet.Layer(layers.LayerTypeDNS).(*layers.DNS)
	if !ok {
```